cmd/rookctl/object/create.go

```diff
@@ -17,43 +17,91 @@ package object
 
 import (
 	"fmt"
+	"io/ioutil"
 	"os"
+	"time"
 
+	"github.com/rook/rook/cmd/rookctl/pool"
 	"github.com/rook/rook/cmd/rookctl/rook"
+	"github.com/rook/rook/pkg/model"
 	"github.com/rook/rook/pkg/rook/client"
+	"github.com/rook/rook/pkg/util/flags"
 	"github.com/spf13/cobra"
 )
 
-var createCmd = &cobra.Command{
-	Use:   "create",
-	Short: "Creates a new object storage instance in the cluster",
-}
+const (
+	initObjectStoreTimeout = 60
+)
+
+var (
+	store           model.ObjectStore
+	dataConfig      pool.Config
+	metadataConfig  pool.Config
+	certificateFile string
+	createCmd       = &cobra.Command{
+		Use:   "create",
+		Short: "Creates a new object storage instance in the cluster",
+	}
+)
 
 func init() {
+	createCmd.Flags().StringVarP(&store.Name, "name", "n", "default", "The name of the object store instance")
+	createCmd.Flags().Int32VarP(&store.Port, "port", "p", model.RGWPort, "The port on which to expose the object store")
+	createCmd.Flags().Int32Var(&store.RGWReplicas, "rgw-replicas", 1, "The number of RGW services for load balancing")
+	createCmd.Flags().StringVar(&certificateFile, "certificate", "", "Path to the ssl cert file (pem format)")
+	pool.AddPoolFlags(createCmd, "data-", &dataConfig)
+	pool.AddPoolFlags(createCmd, "metadata-", &metadataConfig)
+
 	createCmd.RunE = createObjectStoreEntry
 }
 
 func createObjectStoreEntry(cmd *cobra.Command, args []string) error {
 	rook.SetupLogging()
+	if err := flags.VerifyRequiredFlags(cmd, []string{"name", "data-type", "metadata-type"}); err != nil {
+		return err
+	}
+
+	if certificateFile != "" {
+		cert, err := ioutil.ReadFile(certificateFile)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "failed to read certificate", err)
+			os.Exit(1)
+		}
+		store.Certificate = string(cert)
+	}
+
+	dataPool, err := pool.ConfigToModel(dataConfig)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+	store.DataConfig = *dataPool
+	metadataPool, err := pool.ConfigToModel(metadataConfig)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+	store.MetadataConfig = *metadataPool
 
-	c := rook.NewRookNetworkRestClient()
-	out, err := createObjectStore(c)
+	c := rook.NewRookNetworkRestClientWithTimeout(initObjectStoreTimeout * time.Second)
```

Comment:
does the create happen synchronously now? previously, it was an async operation so no extended timeout would be needed here for Create.  Later operations, such as connection/user no longer have the extended timeout they used to, so I'm wondering how the flow works here now.

Reply:
correct, create is synchronous now. All the pools are created before this call returns, so there is no delay later for the other calls.